### ISSUE TITLE
[HOLD for payment 2025-02-04] [$250] Update the Group Creation flow

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#54460]
### Problem

Coming from [this discussion](https://expensify.slack.com/archives/C03U7DCU4/p1734617936214879), let's update the group chat creation flow to remove the ambiguity. Right now, users have to keep clicking exactly to the `Add to Group` button to keep adding people to the group, but if they misclick out of the button to the row, the DM is opened and the progress on group creation is lost.

### Solution

Let's update the flow such that:
- Once the first group member is selected with the `Add to Group` button, the button remains, but clicking the entire row selects or deselects the user for the group (the circle with checkmark indicates the user is selected for the group)
- If no user is selected, clicking the row opens/ creates a DM as it does now
- if at least one user is selected for the group chat, you cannot create a DM
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021872417451020324244</li>
        <li>Upwork Job ID: 1872417451020324244</li>
        <li>Last Price Increase: 2025-01-02</li>
<li>Automatic offers: </li>
<ul>
<li>FitseTLT | Contributor | 105595571</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @VictoriaExpensify</details>

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Update the Group Creation flow

### What is the root cause of that problem?

Currently, if we click on the row, we will always navigate to the DM chat


https://github.com/Expensify/App/blob/5f65aecfccc04483bb4bf07852ecf07ee05e6081/src/pages/NewChatPage.tsx#L194-L197

We only display the check mark if the row is selected
https://github.com/Expensify/App/blob/5f65aecfccc04483bb4bf07852ecf07ee05e6081/src/pages/NewChatPage.tsx#L241
### What changes do you think we should make in order to solve the problem?
1. When we click on a row, if `selectedOptions` is an empty array, navigate to the DM chat. If not we will deselect/select this options

https://github.com/Expensify/App/blob/5f65aecfccc04483bb4bf07852ecf07ee05e6081/src/pages/NewChatPage.tsx#L194-L197


2. If `selectedOptions` is an empty array we will display the `Add to group` button. Otherwise, display the circle with/without a checkmark based on the `item.isSelected`

https://github.com/Expensify/App/blob/5f65aecfccc04483bb4bf07852ecf07ee05e6081/src/pages/NewChatPage.tsx#L241

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)



**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by mountiny** [comment]
This is not open to the proposals yet

**Comment by mountiny** [comment]
Updated the OP

**Comment by mkzie2** [comment]
@mountiny That means we will display both `Add to group` button and check mark when the user is selected, right?

**Comment by mkzie2** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/54460#issuecomment-2559527861)

**Comment by hungvu193** [comment]
Will wait for confirmation before reviewing proposals.

**Comment by mountiny** [comment]
No, when the user is selected the add to group button wont show as they already selected, we will only show the checkmark circle on such row.

**Comment by mkzie2** [comment]
@mountiny Then if this user is de-selected we will show the `Add to group` button, right? Or we will display the circle without a checkmark.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Update the Group Creation flow

### What is the root cause of that problem?
Feature Improvement
### What changes do you think we should make in order to solve the problem?

We will just take the toggleOption [here](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/NewChatPage.tsx#L225-L239) above [createChat](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/NewChatPage.tsx#L194) 
and add a line of code to toggleOption and early return if there are selected options in [createChat](https://github.com/FitseTLT/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/NewChatPage.tsx#L200)
```
if (selectedOptions.length) {
                toggleOption(option);
                return;
            }
```
Another problem we will have is what would happen if the user clicks on selfDM
1. open the self DM: no code change needed
2. Don't display selfDM in the list when when there are selected participants (this is more logical), filter out selfDM
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/pages/NewChatPage.tsx#L160
```
            data: selectedOptions.length ? recentReports.filter((option) => !option.isSelfDM) : recentReports,

```
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by mountiny** [comment]
@mkzie2 we would show the button again.

@hungvu193 can you please review the proposals now? Thanks!

**Comment by hungvu193** [comment]
> @hungvu193 can you please review the proposals now? Thanks!

Sure

**Comment by hungvu193** [comment]
@mountiny I wanna clarify this requirement here:

> if at least one user is selected for the group chat, you cannot create a DM

Does that mean when at least one member is selected, clicking the entire row will add that user to the selected options?

**Comment by mkzie2** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/54460#issuecomment-2559527861). Just removed the point 2 since it's working as expected now.

**Comment by hungvu193** [comment]
Still waiting for the confirmation

**Comment by hungvu193** [comment]
@FitseTLT 's [proposal here](https://github.com/Expensify/App/issues/54460#issuecomment-2563962352) LGTM, it requires less changes and can reuse existing logic.

🎀 👀 🎀  C+ reviewed

**Comment by hungvu193** [comment]
> @hungvu193 This logic is what I copied from `toggleOption` and it's easy to reuse it in the PR phrase if we want to do this.

Hey @mkzie2  I'd choose the most completed proposal, even when I consider accepting your opinion, your updated proposal was still after @FitseTLT posted his proposal.

All yours @mountiny

**Comment by FitseTLT** [comment]
> Thanks!
> 
> @FitseTLT what is your eta for the pr?

PR will be ready tomorrow

**Comment by hungvu193** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by VictoriaExpensify** [comment]
Created an issue to add the regression test to TestRail - https://github.com/Expensify/Expensify/issues/467100

**Comment by VictoriaExpensify** [comment]
**Payment Summary:**

C: @FitseTLT paid $250 via Upworks
C+: @hungvu193 owed $250 via NewDot

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193
